extensions/binding/org.eclipse.smarthome.binding.mqttbroker/README.md

```diff
@@ -0,0 +1,82 @@
+* __publickeypin__: If this is set: After the next connection has been successfully established, the public key of the broker is pinned. The connection will be refused if another public key is used. Clear **publickey** to allow a new public key for the next connection attempt. This option can increase security.
+* __certificate__: The certificate hash. If **certificatepin** is set this hash is used to verify the connection. Clear to allow a new certificate pinning on the next connection attempt. If empty will be filled automatically by the next successful connection. An example input would be `CERTSHA256:83F9171E06A313118889F7D79302BD1B7A2042EE0CFD029ABF8DD06FFA6CD9D3` or `CERTSHA256:83:F9:17:1E:06:A3:13:11:88:89:F7:D7:93:02:BD:1B:7A:20:42:EE:0C:FD:02:9A:BF:8D:D0:6F:FA:6C:D9:D3`.
+* __publickey__: The public key hash. If **publickeypin** is set this hash is used to verify the connection. Clear to allow a new public key pinning on the next connection attempt. If empty will be filled automatically by the next successful connection. An example input would be `CERTSHA256:83F9171E06A313118889F7D79302BD1B7A2042EE0CFD029ABF8DD06FFA6CD9D3` or `CERTSHA256:83:F9:17:1E:06:A3:13:11:88:89:F7:D7:93:02:BD:1B:7A:20:42:EE:0C:FD:02:9A:BF:8D:D0:6F:FA:6C:D9:D3`.
+
+## Channels
+
+There are no channels available. The status message and possible reasons for a failed connection can be accessed through thing properties. The broker connection is successful if the thing is ONLINE.
+
+## Full Example
+
+The example first creates a super secure connection to a broker server, which pins the returned certificate and public key. If someone tries a man in the middle attack later on, this broker connection will recognise it and refuse a connection. Be aware that if your brokers certificate changes, you need to remove the connection entry and add it again.
+
+The second connection is a plain, unsecured one. Use this only for local MQTT Brokers.
+
+A third connection uses a username and password for authentication. The credentials are
+send as plain values of the wire, therefore you should only use this with a secure connection.
+
+In a forth connection, the public key pinning is enabled again. This time, a public key hash (SHA256)
+is provided to pin the connection to a specific server.
+
+`mqttConnections.things`:
+
+```xtend
+mqttbroker:broker:mySecureBroker [ url="ssl://192.168.0.42", certificatepin=true, publickeypin=true ]
+mqttbroker:broker:myUnsecureBroker [ url="tcp://192.168.0.41" ]
+
+mqttbroker:broker:myAuthentificatedBroker [ url="ssl://192.168.0.41", username="user", password="password" ]
+
+mqttbroker:broker:pinToPublicKey [ url="ssl://192.168.0.42", publickeypin=true, publickey="9a6f30e67ae9723579da2575c35daf7da3b370b04ac0bde031f5e1f5e4617eb8" ]
+
+```
```

Comment:
Nice comprehensive and informative README. Would you please break it down to one sentence per line? ([...](https://news.ycombinator.com/item?id=4642395))